### PR TITLE
Add metadata output file to each parflow run

### PR DESCRIPTION
**Pull request by vibraphone**
Write a JSON metadata file summarizing the inputs to a run and its output files. This file provides [ParaView](https://github.com/Kitware/paraview) and other post-processing tools a simple way to aggregate data for visualizations and analyses.

### CODE REVIEW

**Comment by vibraphone** [comment]
@reedmaxwell @lecondon @smithsg84 Would you mind reviewing?

File: pfsimulator/parflow_lib/metadata.c [hunk 1]
```diff
@@ -0,0 +1,699 @@
+#include "parflow.h"
```
Comment (smithsg84) [inline pfsimulator/parflow_lib/metadata.c]: Please include GNU license and optional copyright statement if one is needed by your organization.

File: pfsimulator/parflow_lib/metadata.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#ifndef pf_metadata_h
```
Comment (smithsg84) [inline pfsimulator/parflow_lib/metadata.h]: Please include GNU license and optional copyright statement if one is needed by your organization.

Comment (vibraphone) [reply pfsimulator/parflow_lib/metadata.h]: I've pushed a commit with the licenses/copyrights. Sorry it has taken so long.

**Pull request merged**